### ISSUE TITLE
Error: PutLink has mismatched value list! 

### ISSUE THREAD

**Issue by AmeBel** [opencog/atomspace#333]
loading https://github.com/opencog/atomspace/blob/master/examples/guile/get-put.scm#L70 gives

```
Backtrace:
In ice-9/boot-9.scm:
 157: 11 [catch #t #<catch-closure 1a6e8c0> ...]
In unknown file:
   ?: 10 [apply-smob/1 #<catch-closure 1a6e8c0>]
In ice-9/boot-9.scm:
 157: 9 [catch #t #<catch-closure 1a85c00> ...]
In unknown file:
   ?: 8 [apply-smob/1 #<catch-closure 1a85c00>]
   ?: 7 [call-with-input-string "(define remove-thing-ab\n(PutLink\n(DeleteLink\n(EvaluationLink\n(PredicateNode \"some property\")\n(VariableNode \"$x\")))\n(ListLink\n(ConceptNode \"thing A\")\n(ConceptNode \"B-dom-ness\"))))\n" ...]
In ice-9/boot-9.scm:
2320: 6 [save-module-excursion #<procedure 17899f0 at ice-9/eval-string.scm:65:9 ()>]
In ice-9/eval-string.scm:
  44: 5 [read-and-eval #<input: string 2a6d1a0> #:lang ...]
  37: 4 [lp (define remove-thing-ab (PutLink (DeleteLink #) (ListLink # #)))]
In ice-9/eval.scm:
 453: 3 [eval # ()]
In unknown file:
   ?: 2 [cog-new-link 66 # #]
In ice-9/boot-9.scm:
 102: 1 [#<procedure 2ad11c0 at ice-9/boot-9.scm:97:6 (thrown-k . args)> C++-EXCEPTION ...]
In unknown file:
   ?: 0 [apply-smob/1 #<catch-closure 1a85b00> C++-EXCEPTION ...]

ERROR: In procedure apply-smob/1:
ERROR: In procedure cog-new-link: PutLink has mismatched value list! (/tmp/atomspace-master/opencog/atoms/core/PutLink.cc:178)
ABORT: C++-EXCEPTION
```

**Comment by misgeatgit** [comment]
Adding VariableList will fix it. I encountered same issue recently. The wiki says the variable list are optional but the code has changed and needs it.

**Comment by linas** [comment]
the change was not intentional ...

**Comment by linas** [comment]
Its a bug, fixed in #335

**Comment by AmeBel** [comment]
Thanks
